test/test_jit.py

```diff
@@ -1157,6 +1157,47 @@ def forward(self, x):
                        .check("return") \
                        .run(str(get_module_method(m, 'conv', 'conv2d_forward').graph))
 
+    def test_insert_quant_dequant_multi_uses(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+
+            def forward(self, x, w0, w1, w2):
+                a = F.conv2d(x, w0)
+                b = F.conv2d(a, w1)
+                c = F.conv2d(a, w2)
+                return b + c
+
+        m = torch.jit.script(M())
+        observer = torch.jit.script(Observer())
+        qconfig_dict = {
+            '':
+            QConfig(
+                activation=observer._c,
+                weight=observer._c)
+        }
+        torch._C._jit_pass_insert_observers(m._c, "forward", qconfig_dict, True)
+        data = torch.randn(1, 3, 10, 10, dtype=torch.float)
+        weight = torch.randn(3, 3, 3, 3, dtype=torch.float)
+
+        get_forward(m._c)(data, weight, weight, weight)
+        torch._C._jit_pass_insert_quant_dequant(m._c, "forward", True)
+
+        assert len(m._modules._c.items()) == 0, \
+            'Expected to have zero submodule of conv'
+
+        get_forward(m._c)(data, weight, weight, weight)
+        FileCheck().check("aten::dequantize") \
+                   .check("aten::conv2d") \
+                   .check("aten::quantize_per_tensor") \
+                   .check("aten::dequantize") \
+                   .check_next("aten::dequantize") \
+                   .check("aten::conv2d") \
+                   .check("aten::conv2d") \
+                   .check("return") \
```

Comment:
It is hard to tell what we're checking here. Do we want to make sure that we inserted dequantize for all `conv`s? If that's the case, `FileCheck().check_count` might be better. Also, please note that the IR would probably also contain q-dq nodes for the activations - they already might be screwing up this test. Requiring statements to go one right after another with `check_next` also contributes to the test's brittleness. There is no hard requirement to put these nodes next to each other, they just need to be placed before the first use of their output, so ideally we should not be relying on it.

Reply:
I will just use the check_count to count the number of dequantize, since we expect all the activation input has its own dequant.